### PR TITLE
Pre-commit and CI additions

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Tooling for easier format consistency.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -78,11 +79,21 @@ build-backend = "poetry.core.masonry.api"
...
+in-place = true
+recursive = true
 
 [tool.isort]
-py_version = 38
-profile = "wemake"
+py_version = 39
+profile = "black"
 line_length = 92
+known_first_party = ["qat"]
```
Comment (hamidelmaazouz) [inline pyproject.toml:89]: we could also add these options as `args : []` in the pre-commit config file

File: pyproject.toml [hunk 2]
```diff
@@ -78,11 +79,21 @@ build-backend = "poetry.core.masonry.api"
 remove-all-unused-imports = true
 ignore-init-module-imports = true
 ignore-pass-after-docstring = true
+in-place = true
+recursive = true
```
Comment (hamidelmaazouz) [inline pyproject.toml:83]: an example for pre-commit hook args for autoflake `args: [ --remove-all-unused-imports, --ignore-init-module-imports, --ignore-pass-after-docstring, --in-place ]`

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -13,6 +13,29 @@ permissions:
...
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Formatting tests
+        run: |
+          poetry install --with dev
+          poetry run black --check .
+          poetry run isort --check .
+          poetry run autoflake --check .
```
Comment (hamidelmaazouz) [inline .github/workflows/build.yaml:36]: tests here also need to use the same flags used in pre-commit config file and pyproject (especially line-length)

File: poetry_scripts.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import os
+
+
+def format_code():
+    os.system("poetry run black .")
+    os.system("poetry run isort .")
+    os.system("poetry run autoflake .")
```
Comment (hamidelmaazouz) [inline poetry_scripts.py:7]: same idea here, we could make sure the same flags are consistent across different locations:
- this script
- pre-commit config
- poetry tooling config
- CI tests

**Review by hamidelmaazouz — APPROVED** [review]
Lovely addition Kajsa ! We could also apply these to test source code as well

File: pyproject.toml [hunk 5]
```diff
@@ -78,11 +79,21 @@ build-backend = "poetry.core.masonry.api"
...
-py_version = 38
-profile = "wemake"
+py_version = 39
+profile = "black"
 line_length = 92
+known_first_party = ["qat"]
+
+[tool.black]
+line-length = 92
+target-version = ["py38", "py39", "py310"]
```
Comment (hamidelmaazouz) [inline pyproject.toml:93]: super minor: I believe black can infer which python version from project settings, if not it does it per file basis

**Review by hamidelmaazouz — APPROVED** [review]

File: poetry_scripts.py (hunk 4 above)
Comment (keriksson-rosenqvist) [reply poetry_scripts.py:7]: I have verified that all these different executions make use of the settings defined in `pyproject.toml`.

Comment (hamidelmaazouz) [reply poetry_scripts.py:7]: thanks

**Pull request merged**